src/simplewallet/simplewallet.cpp

```diff
@@ -3214,6 +3215,41 @@ bool simple_wallet::apropos(const std::vector<std::string> &args)
   return true;
 }
 
+bool simple_wallet::scan_tx(const std::vector<std::string> &args)
+{
+  if (args.empty())
+  {
+    PRINT_USAGE(USAGE_SCAN_TX);
+    return true;
+  }
+
+  // Dedup args and make sure they are hashes
+  std::set<string> txids;
+  for (auto s : args) {
+    if (!string_tools::validate_hex(64, s)) {
+      fail_msg_writer() << tr("Invalid txid specified");
+      return true;
+    }
+    txids.insert(s);
+  }
+
+  message_writer() << tr("This operation may reveal the txids to the remote node and affect your privacy in case of using an untrusted node");
+  if (!command_line::is_yes(input_line("Do you want to continue?", true))) {
+    message_writer() << tr("You have canceled the operation");
+    return true;
+  }
+
+  auto saved_m_in_manual_refresh = m_in_manual_refresh.load();
```

Comment:
Use `exchange` instead

Reply:
Btw, i don't think you should restore the previous `m_in_manual_refresh` value.

Reply:
you mean I should do just a `store(true)` ?

Reply:
And set it to `false` once you finished.

Reply:
https://github.com/monero-project/monero/blob/01fdea56adb2ef967b93a3c6b15d291670650f01/src/simplewallet/simplewallet.cpp#L1359-L1360